agent/structs/config_entry_discoverychain.go

```diff
@@ -807,6 +829,63 @@ func (e *ServiceResolverConfigEntry) Validate() error {
 		return fmt.Errorf("Bad ConnectTimeout '%s', must be >= 0", e.ConnectTimeout)
 	}
 
+	if e.LoadBalancer != nil && e.LoadBalancer.EnvoyLBConfig != nil {
+		ec := e.LoadBalancer.EnvoyLBConfig
+
+		validPolicies := map[string]bool{
+			"":                   true,
+			LBPolicyRandom:       true,
+			LBPolicyRoundRobin:   true,
+			LBPolicyLeastRequest: true,
+			LBPolicyRingHash:     true,
+			LBPolicyMaglev:       true,
+		}
+		if ok := validPolicies[ec.Policy]; !ok {
+			return fmt.Errorf("Bad LoadBalancer policy: %q is not supported", ec.Policy)
+		}
+
+		if ec.Policy != LBPolicyRingHash && ec.RingHashConfig != nil {
```

Comment:
Is it required that if you specify ringhash that you populate the ringhash config (and similar for lease request)?